### PR TITLE
Cinema 25 logger

### PR DESCRIPTION
**Pull request by Vasi1i**
Logger added

### CODE REVIEW

File: src/main/java/com/vkatit/cinema/log/Log.java [hunk 1]
```diff
@@ -0,0 +1,69 @@
+package com.vkatit.cinema.log;
```
Comment (isicju) [inline src/main/java/com/vkatit/cinema/log/Log.java]: Log has to be configured using only configuration  file. Don't write any code for it.

File: src/test/java/com/vkatit/cinema/CinemaApplicationTests.java [hunk 2]
```diff
@@ -1,22 +1,82 @@
...
 
-	@Test
-	void contextLoads() {
-		assertTrue(context != null);
-	}
+    public void comparisonFileNameAndMessage(Level expectedLevel, String expectedMessage) {
+        LocalDate currentDate = LocalDate.now();
+        String expectedFileName = PATH + currentDate + (expectedLevel == Level.SEVERE ? "-error.log" : "-info.log");
+        assertTrue(new File(expectedFileName).exists());
+        boolean found = false;
```
Comment (isicju) [inline src/test/java/com/vkatit/cinema/CinemaApplicationTests.java]: can we don't use found at all?

**Review by isicju — CHANGES_REQUESTED** [review]

File: src/test/java/com/vkatit/cinema/CinemaApplicationTests.java (hunk 2 above)
Comment (Vasi1i) [reply src/test/java/com/vkatit/cinema/CinemaApplicationTests.java]: Yes new tests without boolean variable

File: src/main/java/com/vkatit/cinema/log/Log.java (hunk 1 above)
Comment (Vasi1i) [reply src/main/java/com/vkatit/cinema/log/Log.java]: Done

**Comment by Vasi1i** [comment]
Logger redesigned using only XML config

File: src/test/java/com/vkatit/cinema/CinemaApplicationTests.java [hunk 3]
```diff
@@ -1,22 +1,120 @@
...
+            lines = Files.readAllLines(path);
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+        assertTrue(lines.stream().anyMatch(line -> line.contains(message(logLevel))));
+    }
+
+    private void timeForLoggerToWriteLog() {
+        try {
+            Thread.sleep(1000);
```
Comment (isicju) [inline src/test/java/com/vkatit/cinema/CinemaApplicationTests.java]: better use https://www.baeldung.com/awaitility-testing library

**Comment by isicju** [comment]
add some service that use logger with all scenarios

**Comment by isicju** [comment]
I don't see in tests how it check the requirements that I've put in jira. Cover them first.

**Comment by Vasi1i** [comment]
Added LogService (tests for LogService were not created because the service will not be merged with main branch) 
Requirements “Keep logs by date” and “separated log files” were met in tests

**Comment by Vasi1i** [comment]
Added log output to the console
Apply annotation Log4j2
log.path deleted from application.properties
LogExample deleted

File: src/test/java/com/vkatit/cinema/CinemaApplicationTests.java [hunk 4]
```diff
@@ -1,22 +1,159 @@
...
+            return logFile.isPresent();
+        } catch (IOException e) {
+            e.printStackTrace();
+            return false;
+        }
+    }
+
+    private boolean compareLogFileContent(String logLevel, String logFileName) {
+        String logFileFullPath = LOG_PATH + logFileName;
+        timeForLoggerToWriteLog();
```
Comment (isicju) [inline src/test/java/com/vkatit/cinema/CinemaApplicationTests.java]: awaiting library

**Comment by Vasi1i** [comment]
1. “try-catch” blocks are replaced with “throws”
2. Temporary files are created separately
3. “compare” replaced with “has”
4. hamcrest added

**Review by isicju — APPROVED** [review]

**Pull request merged**